src/components/stopwatch/StopWatchController.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <FreeRTOS.h>
+#include <optional>
+#include <timers.h>
+#include "utility/CircularBuffer.h"
+
+namespace Pinetime {
+  namespace System {
+    class SystemTask;
+  }
+
+  namespace Controllers {
+
+    enum class StopWatchStates { Cleared, Running, Paused };
+
+    struct LapInfo {
+      int number = 0;                // Used to label the lap
+      TickType_t timeSinceStart = 0; // Excluding pauses
+    };
+
+    class StopWatchController {
+    public:
+      StopWatchController();
+
+      // StopWatch functionality and data
+      void Start();
+      void Pause();
+      void Clear();
+
+      TickType_t GetElapsedTime();
+
+      // Lap functionality
+
+      /// Only the latest histSize laps are stored
+      void AddLapToHistory();
+
+      /// Returns maxLapNumber
+      int GetMaxLapNumber();
+
+      /// Indexes into lap history, with 0 being the latest lap.
+      std::optional<LapInfo> GetLapFromHistory(int index);
+
+      bool IsRunning();
+      bool IsCleared();
+      bool IsPaused();
+
+    private:
+      // Time at which stopwatch wraps around to zero (1000 hours)
+      TickType_t elapsedTimeBoundary = configTICK_RATE_HZ * 60 * 60 * 1000;
```

Comment:
Can this be constexpr?

Reply:
Fixed. Declaring the expression as `static constexpr` caused it to be interpreted as signed, resulting in an integer overflow error. I could either cast it or change `configTICK_RATE_HZ` from `1024` to `1024u`. The latter would have been technically correct but I didn't know what side effects it could cause so I chose the cast.

Reply:
Interestingly my IDE says it becomes an unsigned value, but perhaps the negative integer result is then getting reinterpreted as unsigned? Not sure, but I agree that casting explicitly is a good idea nonetheless

Could you change this to use `static_cast` rather than a C-style cast?

Reply:
Done.